scripts/update.py

```diff
@@ -0,0 +1,235 @@
+import os
+import platform
+import shutil
+import subprocess
+import sys
+import tarfile
+import tempfile
+import time
+import urllib.parse
+import zipfile
+from enum import StrEnum, auto
+
+import pgpy
+import requests
+from requests import Response
+
+from scripts.progress_bar_updater import UIUpdateProgressBar
+from scripts.utility import quit
+from scripts.version import get_version_info
+
+use_proxy = False  # Set this to True if you want to use a proxy for the update check. Useful for debugging.
+
+
+class UpdateChannel(StrEnum):
```

Comment:
After I saw you did this I also wanted to use StrEnum on my Inhertiance stuff, but it didn't work, so I tested to start the game. 
![image](https://user-images.githubusercontent.com/110975120/230771234-98655fe8-e906-45c5-8eb3-bdf40ea1833f.png)

Do I need to use poetry? poetry.lock used python version 3.11 but in your requirements, we don't have that constraint.

Reply:
Nope, the issue here is that I've been negligent and used StrEnum without checking the supported version. It's only available on 3.11+

I've fixed this by including the StrEnum package as a drop-in replacement.